src/main/java/io/kestra/plugin/git/SyncFlow.java

```diff
@@ -0,0 +1,125 @@
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.DefaultRunContext;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.services.FlowService;
+import io.kestra.plugin.git.services.GitService;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jgit.api.Git;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@SuperBuilder(toBuilder = true)
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Sync a single flow from a Git repository to a Kestra namespace."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Sync a single flow from a Git repository to a specific namespace. This can be used to deploy or update individual flows.",
+            full = true,
+            code = """
+                id: sync_single_flow_from_git
+                namespace: system.cicd
+
+                tasks:
+                  - id: sync_my_flow
+                    type: io.kestra.plugin.git.SyncFlow
+                    url: https://github.com/my-org/kestra-flows # required
+                    branch: main # required
+                    username: my_git_username
+                    password: "{{ secret('GITHUB_ACCESS_TOKEN') }}" # required
+                    targetNamespace: dev.marketing
+                    flowPath: "flows/marketing/flow.yml"
+                """
+        )
+    }
+)
+public class SyncFlow extends AbstractGitTask implements RunnableTask<SyncFlow.Output> {
+
+    public static final Pattern NAMESPACE_FINDER_PATTERN = Pattern.compile("(?m)^\\s*namespace:\\s*(.*)$");
+
+    @Schema(title = "The branch to clone from.")
+    @Builder.Default
+    private Property<String> branch = Property.of("main");
+
+    @Override
+    public Property<String> getBranch() {
+        return this.branch;
+    }
+
+    @Schema(title = "The target namespace where the flow will be synced.")
+    @NotNull
+    private Property<String> targetNamespace;
+
+    @Schema(title = "The full path to the flow YAML file within the Git repository.")
+    @NotNull
+    private Property<String> flowPath;
+
+    @Schema(title = "If true, the task will only log the action without actually syncing the flow.")
+    @Builder.Default
+    private Property<Boolean> dryRun = Property.of(Boolean.FALSE);
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        GitService gitService = new GitService(this);
+        FlowService flowService = ((DefaultRunContext) runContext).getApplicationContext().getBean(FlowService.class);
```

Comment:
We started to discuss with @loicmathieu about avoiding to couple too strongly plugins with Kestra core so not very sure it's a good idea to use directly `FlowService` here.